### ISSUE TITLE
Executable files are not executable (`+x`) after upgrade

### ISSUE THREAD

**Issue by pjrobertson** [nextcloud/vm#2619]
### Steps To Reproduce

Every time I upgrade using `sudo bash /var/scripts/menu.sh` my executable files (mainly: `notify_push` and `translate/bin/node` are not executable which causes errors. If I manually `chmod +x` these files then it works OK.

1. Ugrade a major or minor version using `sudo bash /var/scripts/menu.sh` and going to minor/major upgrade
2. Make sure during this upgrade that either the translations or notify_push apps are updated
3. For translations node: go to the admin settings and see an error stating that there was an error initiating the app, due to node not being executable. For notify_push: run the command in debug mode and see:

```
sudo -u www-data php /var/www/nextcloud/occ notify_push:log debug
sudo journalctl -eu notify_push
...
Mar 30 08:09:41 XX systemd[47543]: notify_push.service: Failed at step EXEC spawning /var/www/nextcloud/apps/notify_push/bin/x86_64/notify_push: Permissio>
```

This *might* be an issue with running the upgrade script with `sudo`, but the script complains if you don't do this.

### Expected Result

The apps installed during an update should be executable and continue to run

### Actual Result

Files are not executable

### Build Version

28.0.4

### Environment

By downloading the VM

**Comment by enoch85** [comment]
Thanks for posting your issue!

Exactly where are these files located? 

I'm thinking, maybe we should add some more files to setup_secure_nextcloud_permissions...?

**Comment by pjrobertson** [comment]
These are the two files I have come across so far that have this issue:

Translate app: `/var/www/nextcloud/apps/translate/bin/node`
notify push app: `/var/www/nextcloud/apps/notify_push/bin/x86_64/notify_push`

**Comment by enoch85** [comment]
Aah, I see what you mean.

For notify push we can implement this:

```
if nextcloud_occ app:list | grep -q notify_push
then
    chmod +x /var/www/nextcloud/apps/notify_push/bin/x86_64/notify_push
    systemctl restart notify_push.service
fi
```

But I'm not sure about the translate app, can you do a `nextcloud_occ app:list | grep translate`?

**Comment by pjrobertson** [comment]
It's the translate app: https://apps.nextcloud.com/apps/translate

```
ncadmin@box:/var/www/nextcloud$ sudo -u www-data php occ app:list | grep translate
  - translate: 2.1.0
```

**Comment by enoch85** [comment]
Just wrote a fix for all files from the top of my head, need to test before merge.

**Comment by enoch85** [comment]
Please test this file: https://raw.githubusercontent.com/nextcloud/vm/2619/nextcloud_update.sh 

It should work with all executable, no matter the name. :)

**Comment by enoch85** [comment]
> After the upgrade, my app also gets upgraded and in the new 2.0 version of the app, my executable is as "great_app/bin/executable2.0"

That's not likely, I don't see *any* reason to change name of the executable, and if that occurs we will handle that then.

I haven't tested this widely, so please run the new script to see if it does what 's expected.

**Comment by pjrobertson** [comment]
That sounds reasonable, yeah. I am probably being overly cautious.

For the script - can I just run it straight off or do I need to wait for an actual upgrade/release?

I found another executable that popped up in my logs just now:

`/var/www/nextcloud/apps/recognize/bin/node`

**Comment by enoch85** [comment]
> For the script - can I just run it straight off or do I need to wait for an actual upgrade/release?

Yes, just run it, it will be like a standard run with the fix implemented. 👍 It won't actually execute any `chmod +x` since that only happens after an actual Nextcloud upgrade.

Though, if you run it with `bash -x` you will be able to see if the variable finds the executables. :)

**Comment by enoch85** [comment]
Were you able to test? :)

**Comment by pjrobertson** [comment]
No not yet, I've had a busy few days! Will try to look at it in the next 2-3 days and reply here 🗡️

**Comment by Reis-A** [comment]
Thank you for this! I had no problems with upgrade to 28.0.5
This finally closes  #2578 which was the same issue., pretty much.
